File: Python/Dosen.py
''' 
	Saya Ibnu Adeng Kurnia NIM 2101769 mengerjakan latihan ke-4 
	dalam mata kuliah desain dan pemrograman berorientasi objek C2 2023
	untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. 
	Aamiin.
'''

'''  
    Design ini disusun dan/atau digunakan hanya untuk lingkungan sendiri.
	Tidak untuk menjadi konsumsi/kepentingan umum.
	Hanya untuk melengkapi tugas DPBO 2023.
'''

from SivitasAkademik import SivitasAkademik

# Membuat kelas dengan nama Dosen #
class Dosen(SivitasAkademik):
    # Atribut kelas private 
    # __nip = ""
    # __pend_terakhir = ""
    # __keahlian =""
    # __pangkatGolongan =""
    # __jabatanFungsional =""

    ## Konstruktor ##
    def __init__(self, nik="-", nama="-", jenis_kelamin='-', asal_universitas="-", email_edu="-", fakultas="-", nip="-", pend_terakhir="-", keahlian="-", pangkatGolongan="-", jabatanFungsional="-"):
        super().__init__(nik, nama, jenis_kelamin, asal_universitas, email_edu, fakultas)
        self.__nip = nip
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian
        self.__pangkatGolongan = pangkatGolongan
        self.__jabatanFungsional = jabatanFungsional

    ####### SETTER DAN GETTTER #######
    ## set dan get untuk nomor induk pegawai
    def setNip(self, nip):
        self.__nip = nip

    def getNip(self):
        return self.__nip

    # set dan get untuk pendidikan terakhir
    def setPend_terakhir(self, pend_terakhir):
        self.__pend_terakhir = pend_terakhir

    def getPend_terakhir(self):
        return self.__pend_terakhir
	
    # set dan get untuk keahliaan
    def setKeahlian(self, keahlian):
        self.__keahlian = keahlian
	
    def getKeahlian(self):
        return self.__keahlian
    
    # set dan get untuk pangkat/golongan
    def setPangkatGolongan(self, pangkatGolongan):
        self.__pangkatGolongan = pangkatGolongan
	
    def getPangkatGolongan(self):
        return self.__pangkatGolongan
    
    # set dan get untuk jabatan fungsional
    def setJabatanFungsional(self, jabatanFungsional):
        self.__jabatanFungsional = jabatanFungsional
	
    def getJabatanFungsional(self):
        return self.__jabatanFungsional

File: C++/program/dosen.cpp
/*  
	Saya Ibnu Adeng Kurnia NIM 2101769 mengerjakan latihan ke-4 
	dalam mata kuliah desain dan pemrograman berorientasi objek C2 2023
	untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. 
	Aamiin.
*/

/*  Design ini disusun dan/atau digunakan hanya untuk lingkungan sendiri.
	Tidak untuk menjadi konsumsi/kepentingan umum.
	Hanya untuk melengkapi tugas DPBO 2023.
*/

#include "header.hh"

// ------------------------ // ^_^ Membuat kelas dengan nama dosen yang terhubung dengan SivitasAkademika  ^_^ // ------------------------ //
// ---- kelas dosen merupakan anak kelas dari SivitasAkademik ------ // 
// Pada intinya bahwa class dosen merupakan anak dari kelas sivitas akademik. Mengapa? Ya tentu karena dilihat 
// dari sisi objek, bahwa class SivitasAkademik dan class dosen merupakan sama-sama manusia
// yang menjalankan tugasnya masing-masing sesuai dengan juknis/juklak/tupoksi-Nya.


// ------- // Constructor // ------- //
Dosen::Dosen(){
    this->NIP = "";					// data NIP //
    this->fakultas = "";			// data fakultas //
    this->pend_terakhir = "";		// data pendidikan terakhir //
    this->keahlian = "";			// data keahlian //
}

// ------- // konstruktor // ------- //
Dosen::Dosen(string NIK, string NIP, string nama, string jenis_kelamin, string asal_universitas, string email_edu, string fakultas, string pend_terakhir, string keahlian)
    : SivitasAkademik(NIK, nama, jenis_kelamin, asal_universitas, email_edu){
    this->NIP = NIP;
    this->fakultas = fakultas;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

// ------- // setter dan getter // ------- //
// set dan get untuk nomor induk pegawai 
void Dosen::setNIP(string NIP){
    this->NIP = NIP;
}

string Dosen::getNIP(){
    return this->NIP;
}

// set dan get untuk data fakultas 
void Dosen::setFakultas(string fakultas){
    this->fakultas = fakultas;
}
        
string Dosen::getFakultas(){
    return this->fakultas;
}
 
// ------- // set dan get untuk data pendidikan terakhir       
void Dosen::setPendidikanTerakhir(string pend_terakhir){
    this->pend_terakhir = pend_terakhir;
}
        
string Dosen::getPendidikanTerakhir(){
    return this->pend_terakhir;
}

// ------- // set dan get untuk data keahlian
void Dosen::setKeahlian(string keahlian){
    this->keahlian = keahlian;
}

string Dosen::getKeahlian(){
    return this->keahlian;
}

// ------- // Destructor // ------- //
Dosen::~Dosen(){}

